lib/new_relic/agent/new_relic_service.rb

```diff
@@ -426,7 +426,16 @@ def invoke_remote(method, payload = [], options = {})
         size = data.size
 
         # Preconnect needs to always use the configured collector host, not the redirect host
-        endpoint_specific_collector = (method == :preconnect) ? @configured_collector : @collector
+        # endpoint_specific_collector = (method == :preconnect) ? @configured_collector : @collector
+
+        # This is a temporary workaround due to errors occurring on the staging collector. The prod collector does not have the same issue.
+        # The staging collector does not respond correctly when using the configured collector host for preconnect, so must use the redirect host
+        # Once this issue is resolved on the staging collector, use the original line that is commented out above.
+        endpoint_specific_collector = if method == :preconnect && (@configured_collector && @configured_collector.name != 'staging-collector.newrelic.com')
+                                        @configured_collector
```

Comment:
Is there a way to log when this condition is entered for staging?  The concern here is that this if condition will mask the problem and we will not come back to it to remove the if condition.

Could we also create another issue to check on logs after this is release. In essence to visit in x months whether this condition that we a adding is still necessary.

Reply:
I think adding log messages here to which part of the if is being used is a great idea, I'll add that.